safekeeper/src/copy_timeline.rs

```diff
@@ -128,6 +130,7 @@ pub async fn handle_request(
     assert!(first_ondisk_segment >= first_segment);
 
     copy_s3_segments(
+        &storage,
```

Comment:
nit: let's check for `None` here and skip `copy_s3_segments` -- if remote storage isn't configured, all segments will be on local disk.